City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# Uhf 5140 group news alias menu

# Group news parent page and group menu visibility [UHF-5140](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5140)
A longer description of the task

## What was done
* Added group field for news parent page (e.g. news archive).
* Added some missing group related translations.
* Enabled content language for groups. Now the group language can be set e.g. to Finnish so that the UI texts for group editors are shown in Finnish.
* Added token for group news parent path. Now group news' URLs and breadcrumbs can be set under the news archive page.
* Show group menu with every group page. Now the menu is visible also when the group news (or other pages) are not put to the group menu.
* Limit group news views to only include news with current language.
* Added link tab to quickly access group nodes. Check group nodes local tasks next to View, Edit, etc.

## How to install
* Make sure your instance is up and running.
    * `git pull origin UHF-5140-group-news-alias`
    * `make fresh`
* Run `make drush-cim`
* Run `make drush-cr`

## How to test
* Make sure you have some groups.
* Create news archive pages for these groups.
* Set the news parent at group edit page (`/group/{group_id}/edit`).
* Create some group news and make sure that the path and breadcrumb have the news archive section before the news content.
* Make sure that group menu is visible with news pages. Also check the mobile menu.
* Check that the group news archive only contains news with current language.
* Check that the new "Group nodes" tab link works (next to View, Edit, etc. at group nodes).
